[opam 2.1~beta2] opam 1.2 format is not correctly upgraded to opam 2.0 format

```
$ opam --version
2.1.0~beta2
$ opam pin add -yn git://github.com/ocaml/opam.git
$ opam install dune-release
[ERROR] Package conflict!
  * Missing dependency:
    - dune-release → opam-format
    unmet availability conditions, e.g. 'ocaml-version >= "4.02.3"'

No solution found, exiting
```

--- a/master_changes.md
+++ b/master_changes.md
@@ -25,7 +25,7 @@ New option/command/subcommand are prefixed with ◈.
   * Fix `--update-invariant` when removing or changing package name [#4360 @AltGr - fix #4353]
 
 ## Pin
-  *
+  * url pin: fix opamfile format upgrade
 
 ## List
   *

--- a/src/client/opamPinCommand.ml
+++ b/src/client/opamPinCommand.ml
@@ -495,6 +495,7 @@ and source_pin
            (OpamStd.Option.to_string OpamUrl.to_string target_url)
            (OpamStd.Format.itemize (fun x -> x) [err]));
   in
+  let opam_opt = opam_opt >>| OpamFormatUpgrade.opam_file in
 
   let nv =
     match version with

